### DISCUSSION TITLE
Regex comment hints

### DISCUSSION THREAD

**Discussion by bchubb-web** [phpactor/phpactor#2680] [Ideas]
Context: in my work codebase we use static magic methods on our models to make structured DB queries, meaning ::findwhere.*( methods are always valid, it would be very useful to be able to define method comments (im not sure of their proper name) with regex to cover a variety of possible methods e.g:
`/** @method array|static findWhere.*() */`
this would probably be useful for many MVC codebases with magic properties/methods on their models, im not sure if this is even possible but would be a GREAT addition and remove a lot of mess from my codebase

(another simple solution would be that any class with __get/set/call doesn't errors for unknown methods/properties)

**Comment by dantleech** [comment]
for stuff like this I wouldn't implement _more_ than PHPStan or Psalm, which use `@property` for cases like this:

https://phpstan.org/writing-php-code/phpdocs-basics#magic-properties

but wildcards are not supported AFAIK. Does PHPStan also complain in the same way that Phpactor does?
